### ISSUE TITLE
Cycles

### ISSUE THREAD

**Issue by BlackGammon** [mlbright/edmonds#4]
Hello,

I have executed your code and the result I'm getting has the correct amount of edges but it's still full of cycles. Something has probably gone wrong in the detection of cycles.

EO

**Comment by BlackGammon** [comment]
Look at the results with this graph:
G = {0: {1: 100}, 1: {2: 1}, 2: {1: 1}}

(graph of three points with the root far away from the others)
We expect to have (0,1) and (1,2) but we get (1,2) and (2,1)

**Comment by BlackGammon** [comment]
I tried with G = {0: {1: 100.0}, 1: {2: 1.0}, 2: {1: 1.0}} and root=0

it returns g = {0: {1: 1.0}, 1: {2: 1.0}} which means the value in g are not the same as in G

**Comment by BlackGammon** [comment]
I tried with G = {0: {1: 100.0}, 1: {2: 2.0}, 2: {1: 3.0}} and root = 0

it returns g =  {0: {1: 3.0}, 1: {2: 2.0}, 2: {}} which means the values in g are wrong.

**Comment by jungokasai** [comment]
I ran into the same problem.

**Comment by matus-pikuliak** [comment]
Yep, the code is faulty. I found working solution on SO (see David Eisenstat's answer):

https://stackoverflow.com/questions/23988236/chu-liu-edmonds-algorithm-for-minimum-spanning-tree-on-directed-graphs
